toss/trajectory_tools.py

```diff
@@ -34,32 +39,87 @@ def compute_trajectory(x: np.ndarray, args, func: Callable) -> Union[np.ndarray,
-                final_time (float): Final time (in seconds) for the integration of trajectory.
+                start_time (int): Start time (in seconds) for the integration of trajectory.
+                final_time (int): Final time (in seconds) for the integration of trajectory.
                 initial_time_step (float): Size of initial time step (in seconds) for integration of trajectory.
                 radius_bounding_sphere (float): Radius of the bounding sphere representing risk zone for collisions with celestial body.
-                event (int): Event configuration (0 = no event, 1 = collision with body detection)
+                event (int): Event configuration (0 = no event, 1 = collision with body detection).
+                number_of_maneuvers (int): Number of possible maneuvers.
+            mesh:
+                vertices (np.ndarray): Array containing all points on mesh.
+                faces (np.ndarray): Array containing all triangles on the mesh.
+            state: Parameters provided by the state vector
+                time_of_maneuver (float): Time for adding impulsive maneuver [seconds].
+                delta_v (np.ndarray): Array containing the cartesian componants of the impulsive maneuver.
         func (Callable): A function handle for the state update equation required for integration.
 
     Returns:
         trajectory_info (np.ndarray): Numpy array containing information on position and velocity at every time step (columnwise).
         squared_altitudes (float): Sum of squared altitudes above origin for every position
         collision_penalty (bool): Penalty value given for the event of a collision with the celestial body.
     """
-    
-    # Compute trajectory by numerical integration
-    trajectory, trajectory_info = integrate(func, x, args)
+    # Convert osculating orbital elements to cartesian for integration
+    r, v = pk.par2ic(E=x[0:6], mu=args.body.mu)
+    x_cartesian = np.array(r+v)
+
+    # Setup time intervals with/without maneuvers
+    if args.problem.number_of_maneuvers == 0:
+        time_list = [args.problem.start_time, args.problem.final_time]
+    else:
+        # Add state variables related to the impulsive maneuver in args
+        time_list = np.empty((args.problem.number_of_maneuvers+2), dtype=int)
+        delta_v_array = np.empty((3, args.problem.number_of_maneuvers), dtype=np.float64)
+        m = 0
+        time_list[0] = args.problem.start_time
+        time_list[-1] = args.problem.final_time
+        for n in range(6, len(x)-3, 4):
+            time_list[m+1] = int(x[n])
+            delta_v_array[:,m] = [x[n+1], x[n+2], x[n+3]]
+            m += 1
+
+        # Rearrange maneuvers in increasing order of time of execution
+        maneuver_times = time_list[1:-1]
+        correct_order_of_maneuvers = np.argsort(maneuver_times)
+        time_list[1:-1] = maneuver_times[correct_order_of_maneuvers]
+        delta_v_array = delta_v_array[:,correct_order_of_maneuvers]
```

Comment:
so, I guess this means you are optimizing multiple maneuvers by reshuffling. However, I think this unnecessarily increases the complexity of the problem, because if you have some maneuvers [m1,m2] then [m2,m1] is actually the identical solution because it will sort it. However, the chromosomes are not the same so your algorithm doesn't know. Probably it would be better to have a inequality constraints t0 < t1 < t2 in the UDP? This way you reduce the search space significantly?

Reply:
I will open another issue for this.

Reply:
Still open, I think? :) (ideally just link issue with # , e.g. #9  )

Reply:
Here is the issue: #37